chainerx_cc/chainerx/cuda/cuda_device/hyperbolic.cu

```diff
@@ -0,0 +1,103 @@
+#include "chainerx/cuda/cuda_device.h"
+
+#include <cmath>
+#include <cstdint>
+
+#include <cuda_runtime.h>
+
+#include "chainerx/array.h"
+#include "chainerx/cuda/cuda_runtime.h"
+#include "chainerx/cuda/cuda_set_device_scope.h"
+#include "chainerx/cuda/elementwise.cuh"
+#include "chainerx/cuda/numeric.cuh"
+#include "chainerx/cuda/op_regist.h"
+#include "chainerx/device.h"
+#include "chainerx/dtype.h"
+#include "chainerx/numeric.h"
+#include "chainerx/routines/math.h"
+#include "chainerx/scalar.h"
+
+namespace chainerx {
+namespace cuda {
+namespace {
+
+template <typename T>
+struct CudaUnaryOp {
+    T (*func)(T);
+
+    CudaUnaryOp(T (*func)(T)) : func{func} {}
+
+    __device__ void operator()(int64_t /*i*/, CudaType x, CudaType& out) { out = func(x); }
+};
```

Comment:
The above part is slightly different from usual please review.

Reply:
If this looks ok, this struct can be used for other UnaryOps instead of define their own Impl

Reply:
It looks good to me.

Reply:
This struct may be dropped when a simpler way is introduced (like the issue you suggested: #6608) in the near future.

Reply:
Will keep it as it is for now.